File: c/Matrizes/matriz_geral.c
#include <stdio.h>

int main() {

    int N;
    double matriz[10][10];
    double soma = 0;
    int linha, coluna;

    printf("Qual sera a ordem da matriz? ");
    scanf("%d", &N);

    // Leitura da matriz
    for (int i = 0; i < N; i++) {
        for (int j = 0; j < N; j++) {

            printf("Elemento [%d,%d]: ", i, j);
            scanf("%lf", &matriz[i][j]);
        }
    }

    // a) Soma dos positivos
    for (int i = 0; i < N; i++) {
        for (int j = 0; j < N; j++) {

            if (matriz[i][j] > 0) {
                soma += matriz[i][j];
            }
        }
    }

    printf("\nSOMA DOS POSITIVOS: %.1lf\n", soma);

    // b) Linha escolhida
    printf("\nEscolha uma linha: ");
    scanf("%d", &linha);

    printf("LINHA ESCOLHIDA:\n");

    for (int j = 0; j < N; j++) {
        printf("%.1lf ", matriz[linha][j]);
    }

    // c) Coluna escolhida
    printf("\n\nEscolha uma coluna: ");
    scanf("%d", &coluna);

    printf("COLUNA ESCOLHIDA:\n");

    for (int i = 0; i < N; i++) {
        printf("%.1lf ", matriz[i][coluna]);
    }

    // d) Diagonal principal
    printf("\n\nDIAGONAL PRINCIPAL:\n");

    for (int i = 0; i < N; i++) {
        printf("%.1lf ", matriz[i][i]);
    }

    // e) Elevar negativos ao quadrado
    for (int i = 0; i < N; i++) {
        for (int j = 0; j < N; j++) {

            if (matriz[i][j] < 0) {
                matriz[i][j] = matriz[i][j] * matriz[i][j];
            }
        }
    }

    printf("\n\nMATRIZ ALTERADA:\n");

    for (int i = 0; i < N; i++) {
        for (int j = 0; j < N; j++) {

            printf("%.1lf ", matriz[i][j]);
        }
        printf("\n");
    }

    return 0;
}

File: java/Matrizes/MatrizGeral.java
import java.util.Locale;
import java.util.Scanner;

public class MatrizGeral {
    public static void main(String[] args) {

        Locale.setDefault(Locale.US);
        Scanner sc = new Scanner(System.in);

        int N;

        System.out.print("Qual a ordem da matriz? ");
        N = sc.nextInt();

        double[][] matriz = new double[N][N];

        // Leitura da matriz
        for (int i = 0; i < N; i++) {
            for (int j = 0; j < N; j++) {

                System.out.print("Elemento [" + i + "," + j + "]: ");
                matriz[i][j] = sc.nextDouble();
            }
        }

        // a) Soma dos positivos
        double soma = 0;

        for (int i = 0; i < N; i++) {
            for (int j = 0; j < N; j++) {

                if (matriz[i][j] > 0) {
                    soma += matriz[i][j];
                }
            }
        }

        System.out.printf("SOMA DOS POSITIVOS: %.1f%n", soma);

        // b) Linha escolhida
        System.out.print("Escolha uma linha: ");
        int linha = sc.nextInt();

        System.out.println("LINHA ESCOLHIDA:");

        for (int j = 0; j < N; j++) {
            System.out.print(matriz[linha][j] + " ");
        }

        System.out.println();

        // c) Coluna escolhida
        System.out.print("Escolha uma coluna: ");
        int coluna = sc.nextInt();

        System.out.println("COLUNA ESCOLHIDA:");

        for (int i = 0; i < N; i++) {
            System.out.print(matriz[i][coluna] + " ");
        }

        System.out.println();

        // d) Diagonal principal
        System.out.println("DIAGONAL PRINCIPAL:");

        for (int i = 0; i < N; i++) {
            System.out.print(matriz[i][i] + " ");
        }

        System.out.println();

        // e) Elevar negativos ao quadrado
        for (int i = 0; i < N; i++) {
            for (int j = 0; j < N; j++) {

                if (matriz[i][j] < 0) {
                    matriz[i][j] = Math.pow(matriz[i][j], 2);
                }
            }
        }

        System.out.println("MATRIZ ALTERADA:");

        for (int i = 0; i < N; i++) {
            for (int j = 0; j < N; j++) {

                System.out.printf("%.1f ", matriz[i][j]);
            }

            System.out.println();
        }

        sc.close();
    }
}